core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/buffer.go

```diff
@@ -146,20 +137,47 @@ type logEventBuffer struct {
 	// size is the number of blocks supported by the buffer
 	size int32
 
-	numOfLogUpkeeps, fastExecLogsHigh uint32
+	blockRate, logLimit uint32
 	// blocks is the circular buffer of fetched blocks
 	blocks []fetchedBlock
 	// latestBlock is the latest block number seen
 	latestBlock int64
+
+	defaultBlockRate uint32
+	defaultLogLimit  uint32
 }
 
-func newLogEventBuffer(lggr logger.Logger, size, numOfLogUpkeeps, fastExecLogsHigh int) *logEventBuffer {
+func newLogEventBuffer(lggr logger.Logger, size int, chainID *big.Int) *logEventBuffer {
+	var defaultBlockRate, defaultLogLimit uint32
+
+	// identify the default block rate for this chain ID
+	switch chainID.Int64() {
+	case 42161, 421613, 421614: // Arbitrum
+		defaultBlockRate = 4
+	default:
+		defaultBlockRate = 1
+	}
+
+	// identify the default log limit for this chain ID
+	switch chainID.Int64() {
+	case 42161, 421613, 421614: // Arbitrum
+		defaultLogLimit = 1
+	case 1, 4, 5, 42, 11155111: // Eth
+		defaultLogLimit = 20
+	case 10, 420, 56, 97, 137, 80001, 43113, 43114, 8453, 84531: // Optimism, BSC, Polygon, Avax, Base
+		defaultLogLimit = 5
+	default:
+		defaultLogLimit = 1
+	}
```

Comment:
It seems a little messy to use these switch statements in the constructor, but we need a way to identify the appropriate defaults for a given chain ID in two places; when the buffer is first constructed, and later in `SetConfig`, where we fall back on a default if zero values are passed in the config. 

Initially, I had these two switch statements as separate functions that were called in the constructor and in `SetConfig`, but it seemed unnecessary to always run through the switch statements every time `SetConfig` was called. 

I also think this functionality should be internal and accessible only to the log buffer for now; so the original functions had the log buffer as a receiver. This created an additional complexity in the constructor, whereby we'd have to first instantiate the log buffer in the constructor, then call the appropriate default functions on the instance of the buffer, and then return the buffer. 

Instead, I just opted to inline both switch statements in the constuctor to identify the appropriate default values once, and then store those defaults on the buffer instance, alongside the actively used values for block rate and log limit. The `SetConfig` function can then reference those default values on the buffer instance when zero values are passed.

Reply:
Another option could be separate constructors like `newLogEventBufferForEth()`, `newLogEventBufferForArb()`, etc., but that's not ideal for maintenance and thinking of an intuitive name to represent all of `optimism, bsc, polygon, avax, base` but not `arbitrum` won't be easy :P 
Your approach seems reasonable

Reply:
The provider will need it as well, you can create a standalone function that accepts a chain id and returns the defaults